packages/pds/src/services/actor.ts

```diff
@@ -7,12 +7,13 @@ import { DidHandle } from '../db/tables/did-handle'
 import { Record as DeclarationRecord } from '../lexicon/types/app/bsky/system/declaration'
 import { APP_BSKY_GRAPH } from '../lexicon'
 import { actorNotSoftDeletedClause } from '../db/util'
+import { Services } from '.'
 
 export class ActorService {
-  constructor(public db: Database) {}
+  constructor(public services: Services, public db: Database) {}
```

Comment:
I see what you're going for here, giving every service a reference to other services. but we already have an asymmetry in their dependencies (need blobstore/message queue for instance). any reason to not do the same here?

Reply:
this pattern also causes an unfortunate circular dependency. which i generally try to avoid but also have done myself in some places so im not overly worried. can always split out with interfaces if we have the reason to

Reply:
The dependency felt alright to me, but it's coming from the perspective that the service/application layer genuinely is "one thing" that is allowed to call into itself however it wishes.  I see this sort of analogously to how two methods on the same class both share `this`.

The situation that made me reach for this was passing `blobstore` and `messageQueue` into `AdminService` in order to get it to `RecordService`, even though neither of those would be exercised from `AdminService` nor the `RecordService` methods that were being used there.

The dependencies were starting to get leaky, and I saw sharing service creators with each service as a way to sort of acknowledge that and isolate all the dependency plumbing to one place (src/services/index.ts).  At the end of the day it's really no biggie to me— happy to add the dependency plumbing between the services.  Worst case scenario they end-up relying on each other in such a way that they all require the same constructor arguments, which would be fine!

Reply:
Maybe a good in-between here is being explicit about which services each other service relies on?

So for instance: `AdminService` relies on `RecordService` so pass the creator for that into `AdminService` but `RecordService` doesn't rely on `AdminService` so it doesn't need any dependencies. This would just force us to be honest about the dependencies between services. 

I'm not at all worried about it right now, but I think for future development it could help force us to be more honest about our logical dependencies & prevent mixing lower order & higher order services. For instance, it feels wrong to me for `RecordService` (lower order) to be able to call into `FeedService` (higher order) even tho they're both in the service layer. Thoughts? 🧐

Reply:
I'm with ya!  I think I have a decent stand-in for now.  The moderation service, for example, will wire-up its services like this:

```ts
export class ModerationService {
  constructor(
    public db: Database,
    public messageQueue: MessageQueue,
    public blobstore: BlobStore,
  ) {}

  static creator(messageQueue: MessageQueue, blobstore: BlobStore) {
    return (db: Database) => new ModerationService(db, messageQueue, blobstore)
  }

  services = {
    repo: RepoService.creator(this.messageQueue, this.blobstore),
    record: RecordService.creator(this.messageQueue),
  }
```

I think/hope this strikes a balance between making the service dependencies explicit, granular, and isolating the plumbing.

Reply:
sweet that sounds great 👍